sklearn/cluster/_hierarchical_fast.pyx

```diff
@@ -29,15 +25,17 @@ from numpy.math cimport INFINITY
 ###############################################################################
 # Utilities for computing the ward momentum
 
-def compute_ward_dist(cnp.ndarray[DOUBLE, ndim=1, mode='c'] m_1,
-                      cnp.ndarray[DOUBLE, ndim=2, mode='c'] m_2,
-                      cnp.ndarray[INTP, ndim=1, mode='c'] coord_row,
-                      cnp.ndarray[INTP, ndim=1, mode='c'] coord_col,
-                      cnp.ndarray[DOUBLE, ndim=1, mode='c'] res):
-    cdef INTP size_max = coord_row.shape[0]
-    cdef INTP n_features = m_2.shape[1]
-    cdef INTP i, j, row, col
-    cdef DOUBLE pa, n
+cpdef void compute_ward_dist(
+    const cnp.float64_t[::1] m_1,
+    const cnp.float64_t[:, ::1] m_2,
+    const cnp.npy_intp[::1] coord_row,
```

Comment:
It is possible to change all `cnp.npy_intp` to `cnp.intp_t` in this file, please? For instance:
```suggestion
    const cnp.intp_t[::1] coord_row,
```

Reply:
Done!